libs/utils/graph/iso.rs

```diff
@@ -9,8 +9,39 @@ use pathfinding::directed::edmonds_karp::*;
 use petgraph::graph::{Graph, NodeIndex};
 use petgraph::visit::DfsPostOrder;
 use petgraph::{Incoming, Outgoing};
+use serde::{Serialize, Deserialize, Serializer, Deserializer};
 
 // -------------- Shamir Isomorphism Algorithm Helper Functions---------------
+#[derive(Debug, Hash, Eq, PartialEq)] 
+pub struct SetSKey {
+    pub val1: NodeIndex,
```

Comment:
It seems really weird that simple struct like that needs to be serialized. Is the problem that NodeIndex is not serializable?

Reply:
Yeah.  Serde JSON wants all keys of maps to be strings for some reason, so if you leave it as is, it won't serialize since it's a struct.  Previously it was a tuple, and that wasn't serializable as a key either.  So I tried serializing the whole thing differently, but I think that's a bit overkill for now.  All we really need is some way of making NodeIndex tuples into strings.

Reply:
Hmm okay, this is a common use case so I believe there likely is a simple solution to this. I can take a look at this later. Strings make sense because the JSON format is a string format.